test/Qowaiv.UnitTests/YesNoTest.cs

```diff
@@ -716,13 +725,35 @@ public void Explicit_BoolToYesNo_Yes()
         [TestCase(true, "Y")]
         [TestCase(false, "N")]
         [TestCase(null, "?")]
-        public void Explicit_YesNoToBool(bool? expected, string str)
+        public void Explicit_YesNoToNullableBool(bool? expected, YesNo yesNo)
         {
-            var yesNo = YesNo.Parse(str);
             var casted = (bool?)yesNo;
             Assert.AreEqual(expected, casted);
         }
 
+        [TestCase(false, "")]
+        [TestCase(true, "Y")]
+        [TestCase(false, "N")]
+        [TestCase(false, "?")]
+        public void Explicit_YesNoToBool(bool expected, YesNo yesNo)
+        {
+            var casted = (bool)yesNo;
+            Assert.AreEqual(expected, casted);
+        }
+
+        [Test]
+        public void Implicit_InIfStatement()
+        {
+            Assert.Throws<DivideByZeroException>(() =>
```

Comment:
Why a divideByZeroException?

Reply:
It does not really matter, but that is one you don't expect to be thrown by an invalid cast, And you want to be sure, that is reaches the block within the if-statement.